AutoHub.BLL/Services/CarService.cs

```diff
@@ -1,42 +1,102 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
+using AutoHub.BLL.DTOs.CarDTOs;
 using AutoHub.BLL.Interfaces;
 using AutoHub.DAL.Entities;
+using AutoHub.DAL.Enums;
 using AutoHub.DAL.Interfaces;
+using AutoMapper;
 
 namespace AutoHub.BLL.Services
 {
     public class CarService : ICarService
     {
+        private readonly IMapper _mapper;
         private readonly IUnitOfWork _unitOfWork;
 
-        public CarService(IUnitOfWork unitOfWork)
+        public CarService(IUnitOfWork unitOfWork, IMapper mapper)
         {
             _unitOfWork = unitOfWork;
+            _mapper = mapper;
         }
 
-        public IEnumerable<Car> GetAll()
+        public IEnumerable<CarResponseDTO> GetAll()
         {
-            return _unitOfWork.Cars.GetAll();
+            var cars = _unitOfWork.Cars.GetAll();
+            var mappedCars = _mapper.Map<IEnumerable<CarResponseDTO>>(cars);
+            return mappedCars;
         }
 
-        public Car GetById(int id)
+        public CarResponseDTO GetById(int carId)
         {
-            var car = _unitOfWork.Cars.GetById(id);
-            return car;
+            var car = _unitOfWork.Cars.GetById(carId);
+            var mappedCar = _mapper.Map<CarResponseDTO>(car);
+            return mappedCar;
         }
 
-        public Car CreateCar(Car carModel)
+        public void Create(CarCreateRequestDTO createCarDTO)
         {
-            _unitOfWork.Cars.Add(carModel);
+            var car = _mapper.Map<Car>(createCarDTO);
+
+            var brand = _unitOfWork.CarBrands.Find(brand => brand.CarBrandName == createCarDTO.CarBrand)
+                .FirstOrDefault();
+            var model = _unitOfWork.CarModels.Find(model => model.CarModelName == createCarDTO.CarModel)
+                .FirstOrDefault();
+            var color = _unitOfWork.CarColors.Find(color => color.CarColorName == createCarDTO.CarColor)
+                .FirstOrDefault();
+
+            car.CarStatusId = CarStatusEnum.New;
+            car.CarBrand = brand ?? new CarBrand { CarBrandName = createCarDTO.CarBrand };
+            car.CarModel = model ?? new CarModel { CarModelName = createCarDTO.CarModel };
+            car.CarColor = color ?? new CarColor { CarColorName = createCarDTO.CarColor };
+
+            _unitOfWork.Cars.Add(car);
+            _unitOfWork.Commit();
+        }
+
+        public void Update(int carId, CarUpdateRequestDTO updateCarDTO)
```

Comment:
There's too much mapping logic in Create and Update methods. It would be good to move it to a separate class, for example, CarMapper. Inject this class in the constructor

lines 41-51, 61-90 -- two public methods of ICarMapper, and divide them by more private methods